inference-engine/src/transformations/src/ngraph_ops/topk_ie.cpp

```diff
@@ -6,43 +6,45 @@
 
 #include <memory>
 #include <string>
+#include <ngraph/opsets/opset1.hpp>
 
 using namespace std;
 using namespace ngraph;
 
 constexpr NodeTypeInfo op::TopKIE::type_info;
 
-op::TopKIE::TopKIE(const Output<Node>& data, const Output<Node>& k, const int64_t axis, const std::string& mode, const std::string& sort,
-        const Shape& output_shape)
-    : Op({data, k}), axis(axis), mode(mode), sort_type(sort), output_shape(output_shape) {
+
+op::TopKIE::TopKIE(const ngraph::Output<ngraph::Node> &data, const ngraph::Output<ngraph::Node> &k, const int64_t axis, const ngraph::op::TopKMode mode,
+                   const ngraph::op::TopKSortType sort, const ngraph::element::Type &index_element_type)
+    : Op({data, k}), m_axis(axis), m_mode(mode), m_sort_type(sort), m_index_element_type(index_element_type) {
     constructor_validate_and_infer_types();
 }
 
-std::shared_ptr<Node> op::TopKIE::copy_with_new_args(const NodeVector& new_args) const {
-    if (new_args.size() != 2) {
-        throw ngraph_error("Incorrect number of new arguments");
-    }
-
-    return make_shared<TopKIE>(new_args.at(0), new_args.at(1), axis, mode, sort_type, output_shape);
+std::shared_ptr<Node> op::TopKIE::clone_with_new_inputs(const ngraph::OutputVector &new_args) const {
+    check_new_args_count(this, new_args);
+    return make_shared<TopKIE>(new_args.at(0), new_args.at(1), m_axis, m_mode, m_sort_type);
```

Comment:
We should respect `m_index_element_type` or remove it from the Op.

Reply:
finally removed this attribute.